mr/src/main/java/org/apache/iceberg/mr/hive/HiveSchemaUtil.java

```diff
@@ -0,0 +1,171 @@
+   * @return The Iceberg schema
+   */
+  public static Schema schema(String columnNames, String columnTypes, String columnNameDelimiter) {
+    // Parse the configuration parameters
+    List<String> names = new ArrayList<>();
+    Collections.addAll(names, columnNames.split(columnNameDelimiter));
+
+    return HiveSchemaUtil.convert(names, TypeInfoUtils.getTypeInfosFromTypeString(columnTypes));
+  }
+
+  /**
+   * Converts the Hive partition columns to Iceberg identity partition specification.
+   * @param schema The Iceberg schema
+   * @param fieldSchemas The partition column specification
+   * @return The Iceberg partition specification
+   */
+  public static PartitionSpec spec(Schema schema, List<FieldSchema> fieldSchemas) {
+    PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
+    fieldSchemas.forEach(fieldSchema -> builder.identity(fieldSchema.getName()));
+    return builder.build();
+  }
+
+  private static Schema convert(List<String> names, List<TypeInfo> typeInfos) {
+    HiveSchemaVisitor visitor = new HiveSchemaVisitor();
+    return new Schema(visitor.visit(names, typeInfos));
+  }
+
+  private static class HiveSchemaVisitor {
+    private int id;
+
+    private HiveSchemaVisitor() {
+      id = 0;
+    }
+
+    private List<Types.NestedField> visit(List<String> names, List<TypeInfo> typeInfos) {
+      List<Types.NestedField> result = new ArrayList<>(names.size());
+      for (int i = 0; i < names.size(); ++i) {
+        result.add(visit(names.get(i), typeInfos.get(i)));
+      }
+
+      return result;
+    }
+
+    private Types.NestedField visit(String name, TypeInfo typeInfo) {
+      switch (typeInfo.getCategory()) {
+        case PRIMITIVE:
+          switch (((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory()) {
+            case FLOAT:
+              return Types.NestedField.optional(id++, name, Types.FloatType.get());
+            case DOUBLE:
+              return Types.NestedField.optional(id++, name, Types.DoubleType.get());
+            case BOOLEAN:
+              return Types.NestedField.optional(id++, name, Types.BooleanType.get());
+            case BYTE:
+            case SHORT:
+            case INT:
+              return Types.NestedField.optional(id++, name, Types.IntegerType.get());
+            case LONG:
+              return Types.NestedField.optional(id++, name, Types.LongType.get());
+            case BINARY:
+              return Types.NestedField.optional(id++, name, Types.BinaryType.get());
+            case STRING:
+            case VARCHAR:
+              return Types.NestedField.optional(id++, name, Types.StringType.get());
+            case CHAR:
+              Types.FixedType fixedType = Types.FixedType.ofLength(((CharTypeInfo) typeInfo).getLength());
+              return Types.NestedField.optional(id++, name, fixedType);
+            case TIMESTAMP:
+              return Types.NestedField.optional(id++, name, Types.TimestampType.withZone());
```

Comment:
Does Hive support both TIMESTAMP WITH ZONE and WITHOUT ZONE? I thought that the default was WITHOUT ZONE, so I'm surprised to see the default here to with zone.

Reply:
Hive3 has Timestamp and Timestamp with TZ.
https://issues.apache.org/jira/browse/HIVE-14412
https://cwiki.apache.org/confluence/display/Hive/Different+TIMESTAMP+types

My reasoning were that we are better off if we start storing data with TZ information present, so with further upgrades we do not lose precision.

Reply:
Ok. Implemented a different conversion for Hive3/Hive2. So Timestamp will be Timestamp.withoutZone() and TimestampTZ will be Timestamp.withZone() if Hive3 is used.